File: lib/src/models/order.dart
enum OrderStatus { menunggu, diproses, selesai }

class Order {
  final String id;
  final String title;
  final String description;
  final double price;
  OrderStatus status;
  // owner: username or id of the user who created the order.
  // nullable for backwards compatibility with older saved orders.
  final String? owner;

  Order({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    this.status = OrderStatus.menunggu,
    this.owner,
  });
}

File: server/src/models/Order.js
const mongoose = require('mongoose');

const OrderSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    required: true
  },
  price: {
    type: Number,
    required: true,
    min: 0
  },
  status: {
    type: String,
    enum: ['menunggu', 'diproses', 'selesai'],
    default: 'menunggu'
  },
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  ownerUsername: {
    type: String,
    required: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  completedAt: {
    type: Date,
    default: null
  }
});

module.exports = mongoose.model('Order', OrderSchema);
